src/backend/dx12/src/command.rs

```diff
@@ -1231,17 +1294,232 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
+            &ViewInfo {
+                resource: src.resource,
+                kind: src.kind,
+                flags: src.storage_flags,
+                view_kind: image::ViewKind::D2Array, // TODO
+                format: src.dxgi_format,
+                range: image::SubresourceRange {
+                    aspects: format::Aspects::COLOR, // TODO
+                    levels: 0..src.num_levels,
+                    layers: 0..src.kind.num_layers(),
+                },
+            }
+        ).unwrap();
+        unsafe {
+            device.CreateShaderResourceView(src.resource, &srv_desc, srv_handle.cpu);
+            self.raw.SetDescriptorHeaps(1, &mut srv_heap.raw.as_raw());
+        }
+        self.temporary_gpu_heaps.push(srv_heap.raw);
+
+        let filter = match filter {
+            image::Filter::Nearest => d3d12::D3D12_FILTER_MIN_MAG_MIP_POINT,
+            image::Filter::Linear => d3d12::D3D12_FILTER_MIN_MAG_LINEAR_MIP_POINT,
+        };
+
+        struct Instance {
+            rtv: d3d12::D3D12_CPU_DESCRIPTOR_HANDLE,
+            viewport: d3d12::D3D12_VIEWPORT,
+            data: internal::BlitData,
+        };
+        let mut instances = HashMap::<internal::BlitKey, Vec<Instance>>::new();
+        let mut barriers = Vec::new();
+
+        for region in regions {
+            let r = region.borrow();
+
+            let first_layer = r.dst_subresource.layers.start;
+            let num_layers = r.dst_subresource.layers.end - first_layer;
+
+            // WORKAROUND: renderdoc crashes if we destroy the pool too early
+            let rtv_pool = Device::create_descriptor_heap_impl(
+                &mut device.clone(),
+                d3d12::D3D12_DESCRIPTOR_HEAP_TYPE_RTV,
+                false,
+                num_layers as _,
+            );
+            self.rtv_pools.push(rtv_pool.raw.clone());
+
+            let key = match r.dst_subresource.aspects {
+                format::Aspects::COLOR => {
+                    // Create RTVs of the dst image for the miplevel of the current region
+                    for i in 0 .. num_layers {
+                        let mut desc = d3d12::D3D12_RENDER_TARGET_VIEW_DESC {
+                            Format: dst.dxgi_format,
+                            ViewDimension: d3d12::D3D12_RTV_DIMENSION_TEXTURE2DARRAY,
+                            u: unsafe { mem::zeroed() },
+                        };
+
+                        *unsafe { desc.u.Texture2DArray_mut() } = d3d12::D3D12_TEX2D_ARRAY_RTV {
+                            MipSlice: r.dst_subresource.level as _,
+                            FirstArraySlice: (i + first_layer) as u32,
+                            ArraySize: 1,
+                            PlaneSlice: 0, // TODO
+                        };
+
+                        let view = rtv_pool.at(i as _).cpu;
+                        unsafe {
+                            device.CreateRenderTargetView(dst.resource, &desc, view);
```

Comment:
It's probably not easy to see (sorry), but the RTVs only live for the lifetime of one regions. The handles get overriden at each region call (not required anymore after calling OMSetRenderTargets in the previous code).
ATM it will set the wrong RTVs in case of multiple regions

Reply:
Confused myself with my initial implementation.